### PR TITLE
docs: Add upgrade note regarding custom ports

### LINKED ISSUE #15956: expose VXLAN and health ports as a flags
## Proposal / RFE

**Is your feature request related to a problem?**

When deploying Cilium on OpenShift user has to run custom script to modify security groups and enable Cilium ports. This step in the guide is quite error-prone.

OpenShift installer does make the IANA VXLAN port 4789 open, albeit Cilium uses default Linux VXLAN port 8472.

Additionally, having the ability to use a different port would make OpenShift CI setup work out of the box for all providers, presently (https://github.com/openshift/release/pull/17380) Cilium job  is enabled only in Azure (where the NSG rules are more permissive).

Also, the health port 4240 is not settable either, but OpenShift installations usually have ports 9900-9999 open and one of those can should be useable instead of 4240.

**Describe the solution you'd like**

Add a daemon flags and Helm chart values to set VXLAN and health ports, keep the default as is.

### PR DESCRIPTION
**Pull request by errordeveloper**
Close #15956

### CODE REVIEW

**Comment by errordeveloper** [comment]
cc @nathanjsweet @christarazi @michi-covalent FYI

**Review by qmonnet — APPROVED** [review]

**Review by borkmann — APPROVED** [review]

**Comment by borkmann** [comment]
Travis failed with:
```
docker: Error response from daemon: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit.
```

File: Documentation/operations/upgrade.rst [hunk 1]
```diff
@@ -322,6 +322,15 @@ Annotations:
   only considers devices that have a global unicast route in some routing table.
 * The XDP-based prefilter is enabled for all devices specified by ``--devices``
   if ``--prefilter-device`` is set.
+* New flags were added to enable installation where alternative VXLAN/Geneve and
+  health ports need to used (``--tunnel-port`` and ``--cluster-health-port``).
```
Comment (borkmann) [inline Documentation/operations/upgrade.rst]: nit: need to be used

Comment (errordeveloper) [reply Documentation/operations/upgrade.rst]: Thanks!

**Comment by errordeveloper** [comment]
> Travis failed with:
> 
> ```
> docker: Error response from daemon: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit.
> ```

It's know infrastructure issues, we just need to re-run...

**Pull request merged**